### PR TITLE
feat: extract login method to config

### LINKED ISSUE #194: 💡 [REQUEST] - Extract login method to config
### Summary

In `AuthContext` has function `logIn` this uses method default `redirect` 
https://github.com/soofstad/react-oauth2-pkce/blob/b7980288083e99c1b751a2bbae41efc4b94d64cd/src/AuthContext.tsx#L88

Please extract this setting in config so that we can configure it. 

In my application, it is necessary to use the popup method, however, I cannot override this everywhere, since there are calls inside the application. 

My problem arises when the refresh token expires, then the `logIn` function is called already with the `refresh` method.
For example, here:

https://github.com/soofstad/react-oauth2-pkce/blob/b7980288083e99c1b751a2bbae41efc4b94d64cd/src/AuthContext.tsx#L136

I would like to call it like this: 
```js
function handleExpiredRefreshToken(initial = false) {
        // If it's the first page load, OR there is no sessionExpire callback, we trigger a new login
        if (initial)
            return logIn(undefined, undefined, "popup");
        // TODO: Breaking change - remove automatic login during ongoing session
        if (!config.onRefreshTokenExpire)
            return logIn(undefined, undefined, "popup");
        config.onRefreshTokenExpire({
            login: logIn,
            logIn,
        });
    }
  ```
the first condition cannot be redefined in any way, the second one can be used like this: 

```js
const authConfig: TAuthConfig = {
  onRefreshTokenExpire: (event: TRefreshTokenExpiredEvent) => event.logIn(undefined, undefined, "popup"),
}
```
I can also suggest removing the first condition. As a result, the second one will work, which we can override from the configuration

My auth config:

```js
const authConfig = {
  clientId: "01",
  authorizationEndpoint: "auth/authorization",
  tokenEndpoint: "https://example.ru/oauth2/token",
  redirectUri: window.location.origin + "/auth/callback/test",
  logoutEndpoint: "auth/logout",
  onRefreshTokenExpire: (event: TRefreshTokenExpiredEvent) => event.logIn(undefined, undefined, "popup"),
  autoLogin: false,
  state: `${Math.random().toString(36).substring(7)}_${Date.now()}`,
  scope: "offline openid",
} satisfies TAuthConfig;
```

### Basic Example

```js
type TAuthConfig = {
 loginMethod: 'popup' | 'refresh'
}
```

### Drawbacks

The authorization method cannot be controlled

### Unresolved questions

_No response_

### Implementation PR

_No response_

### Reference Issues

_No response_

### PR DESCRIPTION
**Pull request by Myks92**
Resolve #194

### CODE REVIEW

**Review by sebastianvitterso — APPROVED** [review]
I don't see any immediate issues with this!

**Comment by sebastianvitterso** [comment]
Looks like pre-commit isn't happy - biome wants to format. Could you update the PR with these changes?

image

image

**Comment by sebastianvitterso** [comment]
I pushed a commit to make pre-commit happy ✌️

File: src/types.ts [hunk 1]
```diff
@@ -102,6 +103,7 @@ export type TInternalConfig = {
   logoutRedirect?: string
   preLogin?: () => void
   postLogin?: () => void
+  loginMethod?: 'redirect' | 'popup'
```
Comment (soofstad) [inline src/types.ts:106]:
```suggestion
  loginMethod: 'redirect' | 'popup'
```
Since it has a default value, no need to make it optional (possibly null)

**Review by soofstad — COMMENTED** [review]
Looks good! One small comment :)

File: src/types.ts (hunk 1 above)
Comment (Myks92) [reply src/types.ts]: Thanks! Fixed it.

**Review by soofstad — APPROVED** [review]
:+1:

**Comment by soofstad** [comment]
Thank you for your contribution! :tada: :1st_place_medal:

**Comment by Myks92** [comment]
Thanks for the quick approval and quick release of the version!

**Pull request merged**